Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/ProjectStatsController.java

```diff
@@ -0,0 +1,126 @@
+ * This controller has all the mappings required to view the statistics of the
+ * project
+ * 
+ * @author ajaymodi
+ *
+ */
+
+@PropertySource(value = "classpath:/user.properties")
+@Controller
+public class ProjectStatsController {
+
+    @Autowired
+    private IRetrieveProjectManager projectManager;
+
+    @Autowired
+    private INetworkManager networkmanager;
+
+    @Autowired
+    private ProjectStats projectStats;
+
+    @Autowired
+    private Environment env;
+
+    private int getCount(List<IConceptStats> conceptList) {
+
+        int cnt = Integer.parseInt(env.getProperty("project.stats.topcount"));
+        int len = conceptList.size() > cnt ? cnt : conceptList.size();
+        return len;
+    }
+
+    private JSONArray getTopConceptsJson(List<IConceptStats> conceptsList,
+            int length) throws JSONException {
+        JSONArray jsonArray = new JSONArray();
+
+        for (int i = 0; i < length; i++) {
+            JSONObject jsonObject = new JSONObject();
+            IConceptStats conceptStats = conceptsList.get(i);
+            jsonObject.put("conceptId", conceptStats.getConceptId());
+            jsonObject.put("description", conceptStats.getDescription());
+            jsonObject.put("label", conceptStats.getLemma());
+            jsonObject.put("count", conceptStats.getCount());
+            jsonArray.put(jsonObject);
+        }
+        return jsonArray;
+    }
+
+    /**
+     * This method gives the visualization of how often concepts appear in the
+     * networks
+     * 
+     * @author Bharath Srikantan & Ajay Modi
+     * @param projectUnixName
+     *            The project unix name
+     * @param model
+     *            Model
+     * @return view
+     * @throws JAXBException
+     * @throws QuadrigaStorageException
+     */
+    @RequestMapping(value = "sites/{projectUnixName}/statistics", method = RequestMethod.GET)
+    public String showProjectStatistics(
+            @PathVariable("projectUnixName") String projectUnixName, Model model)
+            throws JAXBException, QuadrigaStorageException {
+        IProject project = projectManager
+                .getProjectDetailsByUnixName(projectUnixName);
+
+        if (project == null) {
+            return "auth/accessissue";
+        }
+
+        String projectId = project.getProjectId();
+        List<INetwork> networks = networkmanager
+                .getNetworksInProject(projectId);
+        List<IConceptStats> conceptsWithCount = null;
+
+        if (!networks.isEmpty()) {
+            conceptsWithCount = projectStats.getConceptCount(networks);
+
+            try {
+                JSONArray jArray = null;
+                int cnt = getCount(conceptsWithCount);
+                jArray = getTopConceptsJson(conceptsWithCount.subList(0, cnt),
+                        cnt);
+
+                model.addAttribute("jsonstring", jArray);
+                model.addAttribute("networkid", "\"\"");
+                model.addAttribute("project", project);

```

Comment:
you probably want to add the project to the model in any case (even if an exception was thrown or there are no networks). Otherwise the navigation won't work anymore (it needs an project in the model).